### PR TITLE
crimson/os: do not configure seastar allocator for alien threads

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/os/futurized_store.h [hunk 1]
```diff
@@ -4,9 +4,8 @@
 #pragma once
 
 #include <string>
-#include <unordered_map>
 #include <map>
-#include <typeinfo>
```
Comment (rzarzynski) [inline src/crimson/os/futurized_store.h:9]: Ah, I bet that's a residue of the pre-errorator, throwing-free inspection of `seastar::future`. Thanks for catching this!

**Review by rzarzynski — APPROVED** [review]
LGTM!

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
the FTBFS should be addressed by #37990

**Comment by tchaikov** [comment]
jenkins test make check

**Pull request merged**